examples/linear_model/plot_quantile_regression.py

```diff
@@ -11,100 +12,296 @@
+    loc=0, scale=0.5 + 0.5 * x, size=x.shape[0]
+)
+a = 5
+y_noisy_pareto = y_true + 10 * (rng.pareto(a, size=x.shape[0]) - 1 / (a - 1))
+
+# %%
+# Let's first visualize the dataset as well as the error distribution between
+# the noisy observations and the true generative process.
 import matplotlib.pyplot as plt
 
-from sklearn.linear_model import QuantileRegressor, LinearRegression
-from sklearn.metrics import mean_absolute_error, mean_squared_error
-from sklearn.model_selection import cross_val_score
+_, axs = plt.subplots(
+    nrows=2, ncols=2, figsize=(15, 10), sharex="col", sharey="col"
+)
 
+axs[0, 0].plot(x, y_true, label="True generative process")
+axs[0, 0].scatter(
+    x, y_noisy_normal, color="black", alpha=0.5, label="Noisy observations"
+)
+axs[0, 1].hist(y_true - y_noisy_normal, edgecolor="black")
 
-def plot_points_highlighted(x, y, model_low, model_high, ax):
-    """Plot points with highlighting."""
-    mask = y <= model_low.predict(X)
-    ax.scatter(x[mask], y[mask], c="k", marker="x")
-    mask = y > model_high.predict(X)
-    ax.scatter(x[mask], y[mask], c="k", marker="x")
-    mask = (y > model_low.predict(X)) & (y <= model_high.predict(X))
-    ax.scatter(x[mask], y[mask], c="k")
 
+axs[1, 0].plot(x, y_true, label="True generative process")
+axs[1, 0].scatter(
+    x, y_noisy_pareto, color="black", alpha=0.5, label="Noisy observations"
+)
+axs[1, 1].hist(y_true - y_noisy_pareto, edgecolor="black")
 
-fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 5), sharey=True)
+axs[0, 0].set_title("Dataset with heteroscedastic Normal distributed targets")
+axs[1, 0].set_title("Dataset with asymmetric Pareto distributed target")
+axs[0, 1].set_title(
+    "Residuals distribution for heteroscedastic Normal distributed targets"
+)
+axs[1, 1].set_title(
+    "Residuals distribution for asymmetric Pareto distributed target"
+)
+axs[0, 0].legend()
+axs[1, 0].legend()
+axs[0, 0].set_ylabel("y")
+axs[1, 0].set_ylabel("y")
+axs[1, 0].set_xlabel("x")
+_ = axs[1, 1].set_xlabel("Residuals")
 
-rng = np.random.RandomState(42)
-x = np.linspace(0, 10, 100)
-X = x[:, np.newaxis]
-y = 10 + 0.5 * x + rng.normal(loc=0, scale=0.5 + 0.5 * x, size=x.shape[0])
-y_mean = 10 + 0.5 * x
-ax1.plot(x, y_mean, "k--")
+# %%
+# With the heteroscedastic Normal distributed target, we observe that the
+# variance of the noise is increasing when the value of the feature `x` is
+# increasing.
+#
+# With the asymmetric Pareto distributed target, we observe that the positive
+# residuals are bounded.
+#
+# These types of noisy targets violate one of the key assumption made when one
+# wants to use a :class:`~sklearn.linear_model.LinearRegression`: the noise is
+# expected to be normally distributed with a constant variance.
+#
+# Thus, :class:`~sklearn.linear_model.QuantileRegressor` can be used in this
+# case to estimate the conditional quantile (e.g. median) instead of the
+# conditional mean as in :class:`sklearn.linear_model.QuantileRegressor`.
```

Comment:
This conclusion is not striking. If I want to estimate the (conditional) expectation of Y and if the assumptions of OLS are violated, then `QuantileRegressor` is no alternative at all. OLS, however, can still be used. It *just* won't be the most efficient estimator anymore (if we have homoscedasticity, then Gauss-Markov applies). But it still gives unbiased estimates of the coefficients.

Modeling expectations and quantiles are just separate issues. BTW, quantile regression also has nicer properties for stricter assumptions:smirk:

Reply:
No problem. The short version is:
- Minimising the pinball loss/absolute error gives you the conditional quantile/median.
- Minimizing squared error gives you the conditional mean (or expected value or expectation).

Mean and median are just two distinct properties of a distribution. If the distribution is symmetric, they happen to coincide.
**The first step before fitting any model is to say which property of the (predictive) distribution we want to predict.**

Text suggestion: In this asymmetric setting, the median or different quantiles give additional insights. On top of that, median estimation is much more robust to outliers and heavy tailed distributions.

But note that extreme quantiles are estimated by very view data points. 95% quantile are more or less estimated by the 5% largest values and thus also a bit sensitive outliers.

Reply:
Point taken, I made the changes.